TombEngine/Game/camera.cpp

```diff
@@ -323,6 +324,117 @@ void MoveCamera(GameVector* ideal, int speed)
+	auto dest = Vector3(pos.x, pos.y, pos.z);
+
+	GameVector from = GameVector(dest, camSlotId->RoomNumber);
+	Camera.fixedCamera = true;
+
+	MoveObjCamera(&from, camSlotId, camMeshId, targetItem, targetMeshId);
+	Camera.timer = -1;
+}
+
+
+void MoveObjCamera(GameVector* ideal, ItemInfo* camSlotId, int camMeshId, ItemInfo* targetItem, int targetMeshId)
+{
+	int	speed = 1;
+	//Get mesh1 to attach camera to
+	//Vector3i pos = Vector3i::Zero;
+	auto pos =  GetJointPosition(camSlotId, camMeshId, Vector3i::Zero);
+	//Get mesh2 to attach target to
+	//Vector3i pos2 = Vector3i::Zero;
+	auto pos2 =  GetJointPosition(targetItem, targetMeshId, Vector3i::Zero);
+
+	if (OldCam.pos.Position.x != pos.x ||
+		OldCam.pos.Position.y != pos.y ||
+		OldCam.pos.Position.z != pos.z ||
+		OldCam.targetDistance != Camera.targetDistance ||
+		OldCam.targetElevation != Camera.targetElevation ||
+		OldCam.actualElevation != Camera.actualElevation ||
+		OldCam.actualAngle != Camera.actualAngle ||
+		OldCam.target.x != Camera.target.x ||
+		OldCam.target.y != Camera.target.y ||
+		OldCam.target.z != Camera.target.z ||
+		Camera.oldType != Camera.type ||
+		BinocularOn)
+	{
+		OldCam.pos.Position.x = pos.x;
+		OldCam.pos.Position.y = pos.y;
+		OldCam.pos.Position.z = pos.z;
+		OldCam.targetDistance = Camera.targetDistance;
+		OldCam.targetElevation = Camera.targetElevation;
+		OldCam.actualElevation = Camera.actualElevation;
+		OldCam.actualAngle = Camera.actualAngle;
+		OldCam.target.x = Camera.target.x;
+		OldCam.target.y = Camera.target.y;
+		OldCam.target.z = Camera.target.z;
+		LastIdeal.x = pos.x;
+		LastIdeal.y = pos.y;
+		LastIdeal.z = pos.z;
+		LastIdeal.RoomNumber = ideal->RoomNumber;
+		LastTarget.x = pos2.x;
+		LastTarget.y = pos2.y;
+		LastTarget.z = pos2.z;
+	}
+	else
+	{
+		pos.x = LastIdeal.x;
+		pos.y = LastIdeal.y;
+		pos.z = LastIdeal.z;
+		ideal->RoomNumber = LastIdeal.RoomNumber;
+		pos2.x = LastTarget.x;
+		pos2.y = LastTarget.y;
+		pos2.z = LastTarget.z;
+	}
+
+	Camera.pos.x += (ideal->x - Camera.pos.x) / speed;
+	Camera.pos.y += (ideal->y - Camera.pos.y) / speed;
+	Camera.pos.z += (ideal->z - Camera.pos.z) / speed;
+
+	Camera.pos.RoomNumber = GetCollision(Camera.pos.x, Camera.pos.y, Camera.pos.z, Camera.pos.RoomNumber).RoomNumber;
+	LookAt(&Camera, 0);
+
+	int anglex = Camera.target.x - Camera.pos.x;
+	int angley = Camera.target.y - Camera.pos.y;
+	int anglez = Camera.target.z - Camera.pos.z;
+
+	auto position = Vector3i(Camera.target.x - Camera.pos.x, Camera.target.y - Camera.pos.y, Camera.target.z - Camera.pos.z);
+
+	// write last frame camera angle to LastAngle to compare if next frame camera angle has a bigger step than 100.
+	// To make camera movement smoother a speed of 2 is used.
+	// While for big camera angle steps (cuts) -
+	// the speed is set to 1 to make the cut immediatelly.
+	if (LastTarget.x - Camera.target.x > 100 ||
+		LastTarget.y - Camera.target.y > 100 ||
+		LastTarget.z - Camera.target.z > 100)
```

Comment:
I recommend putting something like `constexpr int angleThresholdDegrees = 100;` in the function and then using that in place of `100` to avoid magic numbers and to make it easier to change later.

Reply:
fixed